bundle/edu.gemini.seqexec.odb/src/main/scala/edu/gemini/seqexec/odb/SeqExecService.scala

```diff
@@ -94,6 +102,19 @@ object SeqExecService {
       ConfigBridge.extractSequence(obs, null, IDENTITY_MAP, true)
     }
 
+  private def extractTargets(obs: ISPObservation): TrySeq[List[(TargetType, Target)]] =
+    catchingAll {
+      ObsContext.create(obs).asScala.headOption.map{te =>
+        val guideTargets = te.getTargets().getGuideEnvironment().getTargets().asScala.map(_.getTarget).toList
+        val allTargets = te.getTargets.getTargets.asScala.map(_.getTarget).toList
+        val userTargets = te.getTargets.getUserTargets.asScala.map(_.target.getTarget).toList
+        val scienceTargets = allTargets.diff(guideTargets).diff(userTargets)
```

Comment:
Could also be `val scienceTargets = te.getAsterism.allTargets` and avoid the `diff`s.

Reply:
Do user targets get included in the asterism?

Reply:
No, they are kept separate in the target environment.